src/main/java/io/swagger/swaggerhub/plugin/services/DefinitionFileFinder.java

```diff
@@ -0,0 +1,62 @@
+package io.swagger.swaggerhub.plugin.services;
+
+import org.apache.commons.io.FilenameUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Service to search for and load definition files contained within downstream projects directory
+ */
+public class DefinitionFileFinder {
+
+    private String[] supportedFileExtensions = {"json", "yml", "yaml"};
+
+    /**
+     * Given a directory and the optional regex pattern, find API definitions.
+     * Definitions should be contained in a file with one of the following extensions: .json, .yml, .yaml.
+     * @param directory
+     * @param fileNameRegexPattern
+     * @return
+     * @throws IOException - Returned when the directory specified cannot be found
+     */
+    public List<File> findDefinitionFiles(String directory, Optional<String> fileNameRegexPattern) throws IOException {
+
+        File directoryFile = new File(directory);
```

Comment:
something to consider here is clean up of files after you are done with them

Reply:
On this point - to guard against what you mentioned (directories being created), I ran the _invalidDirectoryThrowsExceptionTest_ in debug mode to check if the non-existent is created. It wasn't thankfully. Am I looking at this point too simply?